### PR TITLE
Openfin layouts release

### PR DESCRIPTION
**Pull request by msuperina**

### CODE REVIEW

File: src/child/index.js [hunk 1]
```diff
@@ -33,6 +34,8 @@ currentWindowService.ready(() => {
     const store = currentWindow.contentWindow.opener.store;
     const rootElement = document.getElementById('app');
 
+    startLayoutsService(store);
```
Comment (owennw) [inline src/child/index.js]: Given both of these layout services take the store, and look at actions, could they be expressed as middleware and folded into the store creation?

File: src/parent/services/LayoutsService.js [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+const uuid = fin.desktop.Application.getCurrent().uuid;
+
+export function start(store) {
+    const layoutsStore = LayoutsStore();
+
+    fin.desktop.InterApplicationBus.subscribe(uuid, 'join-snap-group', updateGroupsAndNotify);
+    fin.desktop.InterApplicationBus.subscribe(uuid, 'leave-snap-group', updateGroupsAndNotify);
+
+    store.subscribe(updateGroupsAndNotify);
```
Comment (owennw) [inline src/parent/services/LayoutsService.js]: Took me a while to find this function -- would've been easier to locate if this were below the function declaration

File: src/parent/services/LayoutsService.js [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+            {}
+        );
+
+        const nextCounts = sameGroupCounts();
+        return diff(previousCounts, nextCounts);
+    }
+
+    function sameGroupCounts() {
+        return Object.keys(sameGroup || {}).reduce(
+            (counts, key) => (counts[key] = sameGroup[key].length) && counts,
```
Comment (owennw) [inline src/parent/services/LayoutsService.js]: An assignment into an operator :o! Not my personal cup of tea.

File: src/parent/services/LayoutsService.js [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+            {}
+        );
+    }
+
+    function diff(previousCounts, nextCounts) {
+        return {
+            joined: Object.keys(nextCounts).filter(key => nextCounts[key] !== previousCounts[key] && nextCounts[key] > 1),
+            left: Object.keys(nextCounts).filter(key => nextCounts[key] !== previousCounts[key] && nextCounts[key] === 1)
```
Comment (owennw) [inline src/parent/services/LayoutsService.js]: You could store `Object.keys(nextCounts)` above this and only do it once, if you felt the need.

File: src/child/services/LayoutsService.js [hunk 5]
```diff
@@ -0,0 +1,32 @@
+import * as Layouts from 'openfin-layouts';
```
Comment (owennw) [inline src/child/services/LayoutsService.js]: I don't see any unit tests for this file, can you add some?

File: src/parent/services/LayoutsService.js [hunk 6]
```diff
@@ -0,0 +1,70 @@
+import * as Layouts from 'openfin-layouts';
```
Comment (owennw) [inline src/parent/services/LayoutsService.js]: Can you add unit tests for this function?

File: src/child/index.js (hunk 1 above)
Comment (msuperina) [reply src/child/index.js]: I am applying the same pattern as the other services which all get injected the store.
However those service are special: they deal with window layouts, which should not be part of the store.

Comment (owennw) [reply src/child/index.js]: Fair enough!

**Pull request merged**